### PR TITLE
format service parameters as a table

### PR DESCRIPTION
**Pull request by kadel**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind design
/area UX
>
> Documentation changes: CI will automatically skip test runs if it is a doc related change.
> /kind documentation

**What does this PR do / why we need it**:

Currently the operators service parameters are listed as a long list of parameters.

<details>
 <summary>old format</summary>

```
▶ odo catalog describe service postgresql-operator.v0.1.1/Database           
Kind: Database
Version: v1alpha1
Description: Database is the Schema for the the Database Database API
Parameters:
Path   Display
- Path: awsAccessKeyId
  DisplayName: AWS S3 accessKey/token ID
  Description: 'Key ID of AWS S3 storage. Default Value: nil Required to create the
    Secret with the data to allow send the backup files to AWS S3 storage.'
- Path: awsS3BucketName
  DisplayName: AWS S3 Bucket name
  Description: 'Name of AWS S3 storage. Default Value: nil Required to create the
    Secret with the AWS data to allow send the backup files to AWS S3 storage.'
- Path: awsSecretAccessKey
  DisplayName: AWS S3 accessKey/token
  Description: 'Secret/Token of AWS S3 storage. Default Value: nil Required to create
    the Secret with the data to allow send the backup files to AWS S3 storage.'
- Path: awsSecretName
  DisplayName: 'AWS Secret name:'
  Description: 'Name of the secret with the AWS data credentials pre-existing in the
    cluster Default Value: nil See here the template: https://github.com/integr8ly/backup-container-image/blob/master/templates/openshift/sample-config/s3-secret.yaml'
- Path: awsSecretNamespace
  DisplayName: 'AWS Secret namespace:'
  Description: 'Namespace of the secret with the AWS data credentials pre-existing
    in the cluster Default Value: nil NOTE: If the namespace be not informed then
    the operator will try to find it in the same namespace where it is applied'
- Path: configMapDatabaseNameKey
  DisplayName: ConfigMap Database Key
  Description: 'Name of the configMap key where the operator should looking for the
    value for the database name for its env var Default value: nil'
- Path: configMapDatabasePasswordKey
  DisplayName: ConfigMap Password Key
  Description: 'Name of the configMap key where the operator should looking for the
    value for the database user for its env var Default value: nil'
- Path: configMapDatabaseUserKey
  DisplayName: ConfigMap User Key
  Description: 'Name of the configMap key where the operator should looking for the
    value for the database password for its env var Default value: nil'
- Path: configMapName
  DisplayName: ConfigMap name
  Description: 'Name of the ConfigMap where the operator should looking for the EnvVars
    keys and/or values only Default value: nil'
- Path: containerImagePullPolicy
  DisplayName: Container Image Pull Policy
  Description: 'Policy definition to pull the Database Image More info: https://kubernetes.io/docs/concepts/containers/images/
    Default value:'
- Path: databaseCRName
  DisplayName: Name of Database CR
  Description: 'Name of the Database CR applied which this backup will work with Default
    Value: "database"'
- Path: databaseCpu
  DisplayName: Database CPU
  Description: 'CPU resource request which will be available for the database container
    Default value: 10Mi'
- Path: databaseCpuLimit
  DisplayName: Database CPU Limit
  Description: 'Limit of CPU request which will be available for the database container
    Default value: 20Mi'
- Path: databaseMemoryLimit
  DisplayName: Database Memory Limit
  Description: 'Limit of Memory which will be available for the database container
    Default value: 512Mi'
- Path: databaseMemoryRequest
  DisplayName: Database Memory Request
  Description: 'Limit of Memory Request which will be available for the database container
    Default value: 128Mi'
- Path: databaseName
  DisplayName: Database Name
  Description: 'Value for the Database Environment Variable (spec.databaseNameKeyEnvVar).
    Default value: example'
- Path: databaseNameKeyEnvVar
  DisplayName: EnvVar Key (Database Name)
  Description: 'Key Value for the Database Environment Variable in order to inform
    the database mame Note that each database version/image can expected a different
    value for it. Default value: nil'
- Path: databasePassword
  DisplayName: Database Password
  Description: 'Value for the Database Environment Variable (spec.databasePasswordKeyEnvVar).
    Default value: postgres'
- Path: databasePasswordKeyEnvVar
  DisplayName: EnvVar Key (Database Password)
  Description: 'Key Value for the Database Environment Variable in order to inform
    the database password Note that each database version/image can expected a different
    value for it. Default value: nil'
- Path: databaseStorageRequest
  DisplayName: Database Storage Request
  Description: 'Limit of Storage Request which will be available for the database
    container Default value: 1Gi'
- Path: databaseUser
  DisplayName: Database User
  Description: 'Value for the Database Environment Variable (spec.databaseUserKeyEnvVar).
    Default value: postgres'
- Path: databaseUserKeyEnvVar
  DisplayName: EnvVar Key (Database User)
  Description: 'Key Value for the Database Environment Variable in order to inform
    the database user Note that each database version/image can expected a different
    value for it. Default value: nil'
- Path: databaseVersion
  DisplayName: Databaseversion
  Description: 'Database version. (E.g 9.6). Default Value: <9.6> IMPORTANT: Just
    the first 2 digits should be used.'
- Path: encryptKeySecretName
  DisplayName: 'EncryptKey Secret name:'
  Description: 'Name of the secret with the Encrypt data pre-existing in the cluster
    Default Value: nil See here the template: https://github.com/integr8ly/backup-container-image/blob/master/templates/openshift/sample-config/gpg-secret.yaml'
- Path: encryptKeySecretNamespace
  DisplayName: 'EncryptKey Secret namespace:'
  Description: 'Namespace of the secret with the Encrypt data pre-existing in the
    cluster Default Value: nil NOTE: If the namespace be not informed then the operator
    will try to find it in the same namespace where it is applied'
- Path: gpgEmail
  DisplayName: 'Gpg public email:'
  Description: 'GPG email to create the EncryptionKeySecret with this data Default
    Value: nil See here how to create this key : https://help.github.com/en/articles/generating-a-new-gpg-key'
- Path: gpgPublicKey
  DisplayName: 'Gpg public key:'
  Description: 'GPG public key to create the EncryptionKeySecret with this data Default
    Value: nil See here how to create this key : https://help.github.com/en/articles/generating-a-new-gpg-key'
- Path: gpgTrustModel
  DisplayName: 'Gpg trust model:'
  Description: 'GPG trust model to create the EncryptionKeySecret with this data.
    the default value is true when it is empty. Default Value: nil See here how to
    create this key : https://help.github.com/en/articles/generating-a-new-gpg-key'
- Path: image
  DisplayName: Image:tag
  Description: 'Database image:tag Default value: centos/postgresql-96-centos7'
- Path: image
  DisplayName: Image:tag
  Description: 'Image:tag used to do the backup. Default Value: <quay.io/integreatly/backup-container:1.0.8>
    More Info: https://github.com/integr8ly/backup-container-image'
- Path: productName
  DisplayName: AWS tag name
  Description: 'Used to create the directory where the files will be stored Default
    Value: <postgresql>'
- Path: schedule
  DisplayName: Schedule
  Description: 'Schedule period for the CronJob. Default Value: <0 0 * * *> daily
    at 00:00'
- Path: size
  DisplayName: Size
  Description: 'Quantity of instances Default value: 1'

```
</details>

This changes the format and displays all parameters in a table

<details>
 <summary>new format</summary>

```
▶ odo catalog describe service postgresql-operator.v0.1.1/Database         
Kind: Database
Version: v1alpha1
Description: Database is the Schema for the the Database Database API
Parameters:

 PATH                          DISPLAYNAME                     DESCRIPTION                                                                                                       

 awsAccessKeyId                AWS S3 accessKey/token ID       Key ID of AWS S3 storage.                                                                                         
                                                               Default Value: nil Required                                                                                       
                                                               to create the Secret with the                                                                                     
                                                               data to allow send the backup                                                                                     
                                                               files to AWS S3 storage.                                                                                          
 awsS3BucketName               AWS S3 Bucket name              Name of AWS S3 storage.                                                                                           
                                                               Default Value: nil Required to                                                                                    
                                                               create the Secret with the AWS                                                                                    
                                                               data to allow send the backup                                                                                     
                                                               files to AWS S3 storage.                                                                                          
 awsSecretAccessKey            AWS S3 accessKey/token          Secret/Token of AWS S3                                                                                            
                                                               storage. Default Value: nil                                                                                       
                                                               Required to create the Secret                                                                                     
                                                               with the data to allow send                                                                                       
                                                               the backup files to AWS S3                                                                                        
                                                               storage.                                                                                                          
 awsSecretName                 AWS Secret name:                Name of the secret with the AWS data credentials pre-existing                                                     
                                                               in the cluster Default Value: nil See here the template:                                                          
                                                               https://github.com/integr8ly/backup-container-image/blob/master/templates/openshift/sample-config/s3-secret.yaml  
 awsSecretNamespace            AWS Secret namespace:           Namespace of the secret with                                                                                      
                                                               the AWS data credentials                                                                                          
                                                               pre-existing in the cluster                                                                                       
                                                               Default Value: nil NOTE: If                                                                                       
                                                               the namespace be not informed                                                                                     
                                                               then the operator will try to                                                                                     
                                                               find it in the same namespace                                                                                     
                                                               where it is applied                                                                                               
 configMapDatabaseNameKey      ConfigMap Database Key          Name of the configMap key                                                                                         
                                                               where the operator should                                                                                         
                                                               looking for the value for the                                                                                     
                                                               database name for its env var                                                                                     
                                                               Default value: nil                                                                                                
 configMapDatabasePasswordKey  ConfigMap Password Key          Name of the configMap key                                                                                         
                                                               where the operator should                                                                                         
                                                               looking for the value for the                                                                                     
                                                               database user for its env var                                                                                     
                                                               Default value: nil                                                                                                
 configMapDatabaseUserKey      ConfigMap User Key              Name of the configMap key                                                                                         
                                                               where the operator should                                                                                         
                                                               looking for the value for the                                                                                     
                                                               database password for its env                                                                                     
                                                               var Default value: nil                                                                                            
 configMapName                 ConfigMap name                  Name of the ConfigMap where                                                                                       
                                                               the operator should looking                                                                                       
                                                               for the EnvVars keys and/or                                                                                       
                                                               values only Default value: nil                                                                                    
 containerImagePullPolicy      Container Image Pull Policy     Policy definition to pull                                                                                         
                                                               the Database Image More info:                                                                                     
                                                               https://kubernetes.io/docs/concepts/containers/images/                                                            
                                                               Default value:                                                                                                    
 databaseCRName                Name of Database CR             Name of the Database CR                                                                                           
                                                               applied which this backup                                                                                         
                                                               will work with Default Value:                                                                                     
                                                               "database"                                                                                                        
 databaseCpu                   Database CPU                    CPU resource request which                                                                                        
                                                               will be available for the                                                                                         
                                                               database container Default                                                                                        
                                                               value: 10Mi                                                                                                       
 databaseCpuLimit              Database CPU Limit              Limit of CPU request which                                                                                        
                                                               will be available for the                                                                                         
                                                               database container Default                                                                                        
                                                               value: 20Mi                                                                                                       
 databaseMemoryLimit           Database Memory Limit           Limit of Memory which will                                                                                        
                                                               be available for the database                                                                                     
                                                               container Default value: 512Mi                                                                                    
 databaseMemoryRequest         Database Memory Request         Limit of Memory Request which                                                                                     
                                                               will be available for the                                                                                         
                                                               database container Default                                                                                        
                                                               value: 128Mi                                                                                                      
 databaseName                  Database Name                   Value for the Database                                                                                            
                                                               Environment Variable                                                                                              
                                                               (spec.databaseNameKeyEnvVar).                                                                                     
                                                               Default value: example                                                                                            
 databaseNameKeyEnvVar         EnvVar Key (Database Name)      Key Value for the Database                                                                                        
                                                               Environment Variable in                                                                                           
                                                               order to inform the database                                                                                      
                                                               mame Note that each database                                                                                      
                                                               version/image can expected                                                                                        
                                                               a different value for it.                                                                                         
                                                               Default value: nil                                                                                                
 databasePassword              Database Password               Value for the Database                                                                                            
                                                               Environment Variable                                                                                              
                                                               (spec.databasePasswordKeyEnvVar).                                                                                 
                                                               Default value: postgres                                                                                           
 databasePasswordKeyEnvVar     EnvVar Key (Database Password)  Key Value for the Database                                                                                        
                                                               Environment Variable in                                                                                           
                                                               order to inform the database                                                                                      
                                                               password Note that each                                                                                           
                                                               database version/image can                                                                                        
                                                               expected a different value for                                                                                    
                                                               it. Default value: nil                                                                                            
 databaseStorageRequest        Database Storage Request        Limit of Storage Request which                                                                                    
                                                               will be available for the                                                                                         
                                                               database container Default                                                                                        
                                                               value: 1Gi                                                                                                        
 databaseUser                  Database User                   Value for the Database                                                                                            
                                                               Environment Variable                                                                                              
                                                               (spec.databaseUserKeyEnvVar).                                                                                     
                                                               Default value: postgres                                                                                           
 databaseUserKeyEnvVar         EnvVar Key (Database User)      Key Value for the Database                                                                                        
                                                               Environment Variable in                                                                                           
                                                               order to inform the database                                                                                      
                                                               user Note that each database                                                                                      
                                                               version/image can expected                                                                                        
                                                               a different value for it.                                                                                         
                                                               Default value: nil                                                                                                
 databaseVersion               Databaseversion                 Database version. (E.g                                                                                            
                                                               9.6). Default Value: <9.6>                                                                                        
                                                               IMPORTANT: Just the first 2                                                                                       
                                                               digits should be used.                                                                                            
 encryptKeySecretName          EncryptKey Secret name:         Name of the secret with the Encrypt data pre-existing in the cluster Default Value: nil See here the template:    
                                                               https://github.com/integr8ly/backup-container-image/blob/master/templates/openshift/sample-config/gpg-secret.yaml 
 encryptKeySecretNamespace     EncryptKey Secret namespace:    Namespace of the secret with                                                                                      
                                                               the Encrypt data pre-existing                                                                                     
                                                               in the cluster Default Value:                                                                                     
                                                               nil NOTE: If the namespace be                                                                                     
                                                               not informed then the operator                                                                                    
                                                               will try to find it in the                                                                                        
                                                               same namespace where it is                                                                                        
                                                               applied                                                                                                           
 gpgEmail                      Gpg public email:               GPG email to create the EncryptionKeySecret with this                                                             
                                                               data Default Value: nil See here how to create this key :                                                         
                                                               https://help.github.com/en/articles/generating-a-new-gpg-key                                                      
 gpgPublicKey                  Gpg public key:                 GPG public key to create the EncryptionKeySecret with this                                                        
                                                               data Default Value: nil See here how to create this key :                                                         
                                                               https://help.github.com/en/articles/generating-a-new-gpg-key                                                      
 gpgTrustModel                 Gpg trust model:                GPG trust model to create the EncryptionKeySecret with                                                            
                                                               this data. the default value is true when it is empty.                                                            
                                                               Default Value: nil See here how to create this key :                                                              
                                                               https://help.github.com/en/articles/generating-a-new-gpg-key                                                      
 image                         Image:tag                       Database image:tag                                                                                                
                                                               Default value:                                                                                                    
                                                               centos/postgresql-96-centos7                                                                                      
 image                         Image:tag                       Image:tag used to do the backup. Default Value:                                                                   
                                                               <quay.io/integreatly/backup-container:1.0.8>                                                                      
                                                               More Info:                                                                                                        
                                                               https://github.com/integr8ly/backup-container-image                                                               
 productName                   AWS tag name                    Used to create the directory                                                                                      
                                                               where the files will be stored                                                                                    
                                                               Default Value: <postgresql>                                                                                       
 schedule                      Schedule                        Schedule period for the                                                                                           
                                                               CronJob. Default Value: <0 0 *                                                                                    
                                                               * *> daily at 00:00                                                                                               
 size                          Size                            Quantity of instances Default                                                                                     
                                                               value: 1                                                                                                          


```
</details>

**Which issue(s) this PR fixes**:

Fixes #?

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] Update changelog

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by girishramnani** [comment]
so I might need to update the proposal as per this output

**Comment by girishramnani** [comment]
I thought we prefered the list approach?

**Comment by kadel** [comment]
> I thought we prefered the list approach?

When I saw it on console with a long list of parameters it was really hard to undertand

**Comment by girishramnani** [comment]
I am just concerned what would happen when we have one more column called “type” which we are planning to add for swagger and crd support

**Comment by kadel** [comment]
> I am just concerned what would happen when we have one more column called “type” which we are planning to add for swagger and crd support

Then there will be another column. I don't think that that would be a problem unless the new column is something long as a description. 
We can definitely revisit this if adding a new column makes this unreadable. 

This table is also not ideal, but I find it much easier to read than the current list.

**Comment by kadel** [comment]
> And that type and description columns are/would be optional

This might be also nice. I would keep type. But description could be optional and we can add something `--with-description` flag

**Comment by girishramnani** [comment]
> > And that type and description columns are/would be optional
> 
> This might be also nice. I would keep type. But description could be optional and we can add something `--with-description` flag

Yeah this makes more sense

**Comment by feloy** [comment]
A possible output could also be 
```
Parameters:

 PATH / DISPLAYNAME                   DESCRIPTION                                                                                                       

 awsAccessKeyId                       Key ID of AWS S3 storage.                                                                                         
 AWS S3 accessKey/token ID            Default Value: nil Required to create the Secret with the                                                                                     
                                      data to allow send the backup files to AWS S3 storage.                                                                                          
 
 awsS3BucketName                      Name of AWS S3 storage.                                                                                           
 AWS S3 Bucket name                   Default Value: nil Required to create the Secret with the AWS                                                                                    
                                      data to allow send the backup files to AWS S3 storage.                         
```

**Review by dharmit — COMMENTED** [review]
/approve

@feloy please defer LGTM till the discussion here https://github.com/openshift/odo/pull/4812#issuecomment-864119697 is closed.

File: pkg/odo/cli/catalog/describe/operator_backend.go [hunk 1]
```diff
@@ -84,11 +85,26 @@ func (ohb *operatorBackend) RunDescribeService(dso *DescribeServiceOptions) erro
...
 		return nil
 	}
-	output, err := yaml.Marshal(svc.ConvertCRDToRepr(ohb.CR))
-	if err != nil {
-		return err
-	}
+	repr := svc.ConvertCRDToRepr(ohb.CR)
 
-	fmt.Print(string(output))
+	fmt.Printf("Kind: %s\n", repr.Kind)
```
Comment (girishramnani) [inline pkg/odo/cli/catalog/describe/operator_backend.go:90]: can we extract this into a function in this same package?

Comment (girishramnani) [reply pkg/odo/cli/catalog/describe/operator_backend.go:90]: this I meant - https://github.com/openshift/odo/pull/4812/files#diff-0d330dc955dfbc327b5a13bd4933557f7529940e0d7f756d9a186f9349a9d0b6R90-R108

**Comment by girishramnani** [comment]
> A possible output could also be
> 
> ```
> Parameters:
> 
>  PATH / DISPLAYNAME                   DESCRIPTION                                                                                                       
> 
>  awsAccessKeyId                       Key ID of AWS S3 storage.                                                                                         
>  AWS S3 accessKey/token ID            Default Value: nil Required to create the Secret with the                                                                                     
>                                       data to allow send the backup files to AWS S3 storage.                                                                                          
>  
>  awsS3BucketName                      Name of AWS S3 storage.                                                                                           
>  AWS S3 Bucket name                   Default Value: nil Required to create the Secret with the AWS                                                                                    
>                                       data to allow send the backup files to AWS S3 storage.                         
> ```

One the reason I prefer @kadel 's approach over this is because we can refer the path's that need to be set fairly quickly.

File: pkg/odo/cli/catalog/describe/operator_backend.go (hunk 1 above)
Comment (kadel) [reply pkg/odo/cli/catalog/describe/operator_backend.go:90]: you mean in this same package `describe`? Why?

**Comment by kadel** [comment]
> A possible output could also be
> 
> ```
> Parameters:
> 
>  PATH / DISPLAYNAME                   DESCRIPTION                                                                                                       
> 
>  awsAccessKeyId                       Key ID of AWS S3 storage.                                                                                         
>  AWS S3 accessKey/token ID            Default Value: nil Required to create the Secret with the                                                                                     
>                                       data to allow send the backup files to AWS S3 storage.                                                                                          
>  
>  awsS3BucketName                      Name of AWS S3 storage.                                                                                           
>  AWS S3 Bucket name                   Default Value: nil Required to create the Secret with the AWS                                                                                    
>                                       data to allow send the backup files to AWS S3 storage.                         
> ```

I will try to change it to this format to see how it looks when everything is formated like this.

File: pkg/odo/cli/catalog/describe/operator_backend.go (hunk 1 above)
Comment (girishramnani) [reply pkg/odo/cli/catalog/describe/operator_backend.go:90]: keeping it clean?

Comment (kadel) [reply pkg/odo/cli/catalog/describe/operator_backend.go:90]: How it would be cleaner? It would be the same code, only in a different place.

**Comment by kadel** [comment]
> A possible output could also be
> 
> ```
> Parameters:
> 
>  PATH / DISPLAYNAME                   DESCRIPTION                                                                                                       
> 
>  awsAccessKeyId                       Key ID of AWS S3 storage.                                                                                         
>  AWS S3 accessKey/token ID            Default Value: nil Required to create the Secret with the                                                                                     
>                                       data to allow send the backup files to AWS S3 storage.                                                                                          
>  
>  awsS3BucketName                      Name of AWS S3 storage.                                                                                           
>  AWS S3 Bucket name                   Default Value: nil Required to create the Secret with the AWS                                                                                    
>                                       data to allow send the backup files to AWS S3 storage.                         
> ```

To be honest I'm not able to figure out how to achieve this with tablewriter :-/
When I tried merging cells, it looked wired, as there was too much vertical space between path and dispalayname.
When I tired using one cell for path/displayname I couldn't force tablewriter to accept newline "\n", so path and dispalayname ended up on the same line

File: pkg/odo/cli/catalog/describe/operator_backend.go (hunk 1 above)
Comment (girishramnani) [reply pkg/odo/cli/catalog/describe/operator_backend.go:90]: ok, np.

**Comment by girishramnani** [comment]
I am fine with the current format, can we move forward this?

**Comment by mik-dass** [comment]
```
github.com/openshift/odo/pkg/odo/cli/catalog/describe
pkg/odo/cli/catalog/describe/operator_backend.go:101:19: undefined: yaml
make[1]: *** [vet] Error 2
```
/hold

**Comment by kadel** [comment]
> ```
> github.com/openshift/odo/pkg/odo/cli/catalog/describe
> pkg/odo/cli/catalog/describe/operator_backend.go:101:19: undefined: yaml
> make[1]: *** [vet] Error 2
> ```
> 
> 
>     
>   
> 
> /hold

fixed. It was due to conflicting changes in another PR

**Comment by kadel** [comment]
```


failed to acquire lease for "aws-quota-slice": status 500 Internal Server Error, status code 500 | failed to acquire lease for "aws-quota-slice": status 500 Internal Server Error, status code 500
-- | --
failed to acquire lease for "aws-quota-slice": status 500 Internal Server Error, status code 500
10/11 Tests Passed! | expand_less

```

/retest

**Review by dharmit — APPROVED** [review]

**Pull request merged**